### PR TITLE
delete all token related data after token unregistered

### PR DESCRIPTION
**Pull request by zalun**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -160,9 +160,14 @@ app.post('/unregister', function(req, res) {
       throw new Error('Not Found');
     }
 
-    return Promise.all(machines.map(machine => redis.del(machine)));
+    console.log('XXX: deleting ' + token);
+    return Promise.all(machines.map(machine => redis.del(machine)))
+    .then(function() {
```
Comment (marco-c) [inline index.js]: nit: This could be done in parallel with a single Promise.all

File: test/testUnregister.js [hunk 2]
```diff
@@ -46,4 +50,29 @@ describe('mercurius unregister', function() {
...
   });
+
+  it('deletes all token related data after a registration has been removed', function(done) {
+    redis.exists(token)
+    .then(function(exists) {
+      exists.should.equal(0);
+      console.log('ZZZ: checking ' + token);
+      return redis.exists(token + ':clients');
+    })
+    .catch(done)
```
Comment (marco-c) [inline test/testUnregister.js:60]: You can remove all `catch`es and `done`s and simply return the promise. `Mocha` automatically handles promise resolution as a pass and promise rejection as a failure.

**Comment by marco-c** [comment]
Shouldn't we also remove the machine's clients on `unregisterMachine`?

File: test/testUnregister.js (hunk 2 above)
Comment (zalun) [inline test/testUnregister.js:60]: it is hanging and complains of a lack on "done" function

**Comment by zalun** [comment]
Yes - we should. It's gonna be in different PR, I've got some trouble with testing it (multiple nocks and then failures, I'll look for help tomorrow)

File: test/testUnregister.js (hunk 2 above)
Comment (marco-c) [inline test/testUnregister.js:60]: You'd need to do something like:

``` JavaScript
  it('deletes all token related data after a registration has been removed', function() {
    return redis.exists(token)
    .then(function(exists) {
      exists.should.equal(0);
      return redis.exists(token + ':clients');
    })
    .then(function(exists) {
      exists.should.equal(0);
      return redis.exists('machine_1');
    })
    .then(function(exists) {
      exists.should.equal(0);
      return redis.exists('machine_1:clients');
    })
    .then(function(exists) {
      exists.should.equal(0);
    })
  });
```

**Pull request merged**